pkg/networkutils/network.go

```diff
@@ -745,36 +746,28 @@ func setupENINetwork(eniIP string, eniMAC string, eniTable int, eniSubnetCIDR st
 			return errors.Wrap(err, "setupENINetwork: failed to clean up old routes")
 		}
 
-		// In case of route dependency, retry few times
-		retry := 0
-		for {
+		err = retry.RetryNWithBackoff(retry.NewSimpleBackoff(time.Millisecond*500, retryRouteAddInterval*time.Duration(maxRetryRouteAdd), 0.15, 2.0), maxRetryRouteAdd, func() error {
 			if err := netLink.RouteAdd(&r); err != nil {
 				if netlinkwrapper.IsNetworkUnreachableError(err) {
-					retry++
-					if retry > maxRetryRouteAdd {
-						log.Errorf("Failed to add route %s/0 via %s table %d",
-							r.Dst.IP.String(), gw.String(), eniTable)
-						return errors.Wrapf(err, "setupENINetwork: failed to add route %s/0 via %s table %d",
-							r.Dst.IP.String(), gw.String(), eniTable)
-					}
-					log.Debugf("Not able to add route route %s/0 via %s table %d (attempt %d/%d)",
-						r.Dst.IP.String(), gw.String(), eniTable, retry, maxRetryRouteAdd)
+					log.Debugf("Not able to add route route %s/0 via %s table %d ",
+						r.Dst.IP.String(), gw.String(), eniTable, maxRetryRouteAdd)
 					time.Sleep(retryRouteAddInterval)
 				} else if netlinkwrapper.IsRouteExistsError(err) {
 					if err := netLink.RouteReplace(&r); err != nil {
 						return errors.Wrapf(err, "setupENINetwork: unable to replace route entry %s", r.Dst.IP.String())
```

Comment:
Hmm, honestly I'm not sure why we try to Add, then Replace. We should be able to use `netLink.RouteReplace()` right away, since it will also add it if it doesn't already exist. That would make this code even simpler.